### ISSUE TITLE
Lint/SymbolConversion with string interpolation should be a Style cop

### ISSUE THREAD

**Issue by RossBarnie** [rubocop/rubocop#12963]
## Is your feature request related to a problem? Please describe.

The following code shows an example of a violation of the `Lint/SymbolConversion` cop:

```ruby
prefix = "foo"

symbol = "#{prefix}_bar".to_sym # violation here
symbol_valid = :"#{prefix}_bar" # no violation
```

The `Lint/SymbolConversion` cop description says:

> Checks for uses of literal strings converted to a symbol where a literal symbol could be used instead.

This may be a misunderstanding on my part but my assumption is that the `symbol` and `symbol_valid` variables in the above code are functionally equivalent in that the String must be created and interpolated before the symbol can be created. In that case, this case is a style choice and there is no performance improvement by using the "literal" syntax over the `to_sym` method.

## Describe the solution you'd like

I would like:
- `Lint/SymbolConversion` to ignore interpolated strings being converted to a symbol using the `to_sym` method
- A new Style cop created to enforce "always use symbol literals" or "allow to_sym on interpolated strings", exact naming TBD, to give users the choice of which style to use.

## Additional Context

To reiterate, this is based on the assumption that `:"#{thing}"` is equivalent to `"#{thing}".to_sym` in terms of the construction of a string before converting it to a symbol. I am happy to be shown I am wrong on this but I did not see any proofs or justifications in the [original PR that introduced  Lint/SymbolConversion](https://github.com/rubocop/rubocop/pull/9362) nor [the PR that introduced the string interpolation case](https://github.com/rubocop/rubocop/issues/12373).

**Comment by vlad-pisanov** [comment]
FWIW, `.to_sym` is slower on my machine.

```ruby
require 'benchmark/ips'

puts `ruby -v`

bar = "bar"

Benchmark.ips do |x|
  x.report('symbol') { :"foo_#{bar}" }
  x.report('string') { "foo_#{bar}".to_sym }

  x.compare!
end
```

```
ruby 3.3.2 (2024-05-30 revision e5a195edf6) [x86_64-darwin21]
Warming up --------------------------------------
              symbol   622.135k i/100ms
              string   569.997k i/100ms
Calculating -------------------------------------
              symbol      6.104M (± 3.0%) i/s -     31.107M in   5.101083s
              string      5.307M (± 5.9%) i/s -     26.790M in   5.066530s

Comparison:
              symbol:  6103804.6 i/s
              string:  5306964.7 i/s - 1.15x  slower
```

**Comment by RossBarnie** [comment]
I recreated this locally and got 1.10x slower, other benchmarks showed worse performance too. For us this is a rare enough operation that it won't make a significant difference to performance. We don't want to disable the rule because it makes sense to flag `:'some_string'`. I suppose in our case we'll need to disable the rule for the instances where `to_sym` is called on interpolated strings. It would be nice if we could configure the rule to ignore these instead.

**Comment by koic** [comment]
I also agree that this `SymbolConversion` cop is more appropriately categorized as `Style` department than `Lint` department.

**Comment by RossBarnie** [comment]
I think given the discussion above I now agree this should be `Lint`. If we disagree with it we can always disable the rule, closing.
